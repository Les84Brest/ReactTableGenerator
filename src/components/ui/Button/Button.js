import React from "react"
import ClassNames from 'classnames'
import PropTypes from 'prop-types'

export const Button = ({
    children, onClick, className, disabled, active, ...attrs
}) => {

    const classes = ClassNames('btn', className, { active })

    return (
        <button
            className={classes}
            disabled={disabled}
            onClick={onClick}
            {...attrs}
        >{children}</button>
    )
}

Button.propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func,
    className: PropTypes.string,
    disabled: PropTypes.bool
}

Button.defaultProps = {
    children: 'Default button',
    disabled: false,
    onClick: () => { },
    className: '',
    active: false
}