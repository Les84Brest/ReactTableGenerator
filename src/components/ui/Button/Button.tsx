import React, {FC, ReactNode} from "react"
import cn from 'classnames'

interface ButtonProps {
    children?: ReactNode,
    onClick: () => void,
    className?: string,
    disabled?: boolean,
    active?: boolean,
}

export const Button:FC<ButtonProps> = ({
    children, onClick, className, disabled, active, ...attrs
}) => {

    const classes: string = cn('btn', className, { active })

    return (
        <button
            className={classes}
            disabled={disabled}
            onClick={onClick}
            type="button"
            {...attrs}
        >{children}</button>
    )
}

export default Button

