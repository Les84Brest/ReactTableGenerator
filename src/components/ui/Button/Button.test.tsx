import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

const buttonProps = {
    children: 'Press me',
    onClick: () => { },
    className: 'test-class',
    disabled: false,
    active: false,
}

describe('Button tests', () => {

    test('conponent renders', () => {

        render(<Button {...buttonProps} />)
        const btn = screen.getByRole('button');

        expect(btn).toBeInTheDocument();
        expect(btn).toMatchSnapshot();
    })

    test('click works', () => {
        const clickFn = jest.fn();

        const clickButtonProps = { ...buttonProps, onClick: clickFn }
        render(<Button {...clickButtonProps} />);

        const btn = screen.getByRole('button');
        fireEvent.click(btn);
        expect(clickFn).toBeCalledTimes(1);
    })

    test('apply styling', () => {
        const styleButtonProps = { ...buttonProps, className: 'btn-copy' }
        render(<Button {...styleButtonProps} />);
        const btn = screen.getByRole('button');
        expect(btn).toHaveClass('btn-copy');
    });
})

