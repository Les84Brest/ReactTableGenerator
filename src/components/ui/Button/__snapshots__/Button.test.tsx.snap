// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests conponent renders 1`] = `
<button
  class="btn test-class"
  type="button"
>
  Press me
</button>
`;
